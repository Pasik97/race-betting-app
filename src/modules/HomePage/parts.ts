import styled from 'styled-components';

export const WelcomeText = styled.p`
   margin: 0;
   font-size: 18px;
   line-height: 21px;
   text-align: center;
`;
