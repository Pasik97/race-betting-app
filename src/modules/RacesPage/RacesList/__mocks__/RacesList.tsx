const mock = () => '(RacesList /)';

export default mock;
