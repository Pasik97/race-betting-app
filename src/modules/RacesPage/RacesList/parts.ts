import { Link } from "react-router-dom";
import styled from "styled-components";
import media from "utils/media";

export const RacesListWrapper = styled.div``;

export const Cell = styled.div`
   color: #1a1a1a;
   font-size: 16px;
   line-height: 21px;
   border-right: 1px solid #1a1a1a;
   padding: 6px;
   display: flex;
   align-items: center;

   &:last-of-type {
      border-right: none;
   }
`;

export const HeaderCell = styled(Cell)`
   font-weight: 700;
   background-color: #cccccc;
   color: #01263A;
   border-top: 1px solid #1a1a1a;
`;

export const Row = styled.div`
   display: grid;
   grid-template-columns: 0.5fr 2fr 1fr;
   border: 1px solid #1a1a1a;
   border-top: none;

   &:nth-of-type(2n+3) {
      ${Cell} {
         background-color: #eeeeee;
      }
   }

   ${media.fromTablet} {
      display: grid;
      grid-template-columns: 1fr 4fr 1.5fr;
   }

   ${media.fromDesktop} {
      display: grid;
      grid-template-columns: 1fr 6fr 1.5fr;
   }
`;

export const Label = styled.p`
   margin: 0;
   overflow-wrap: anywhere;
`;

export const StyledLink = styled(Link)`
   color: #1a1a1a;
   font-size: 16px;
   line-height: 21px;
   text-decoration: none;

   &:hover {
      color: #01263A;
   }
`;

export const CheckboxWrapper = styled.div`
   display: flex;
   align-items: baseline;
   margin-bottom: 16px;
`;

export const Checkbox = styled.input`
   margin: 0 0 0 16px;
`;
