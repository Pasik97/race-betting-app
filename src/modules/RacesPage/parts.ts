import styled from "styled-components";

export const RacesPageWrapper = styled.div`
    position: relative;
    min-height: 150px;
`;
