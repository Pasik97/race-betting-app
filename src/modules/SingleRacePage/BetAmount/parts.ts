import styled from "styled-components";

export const InputWrapper = styled.div`
   display: flex;
   margin: 24px 0;
`;

export const Label = styled.p`
   margin: 0 8px 0 0;
   font-size: 16px;
   line-height: 18px;
   color: #1a1a1a;
`;

export const Input = styled.input`
   color: #1a1a1a;
   font-size: 16px;
   line-height: 21px;
   padding: 0;
   border: none;
   outline: none;
   border-bottom: 1px solid #dddddd;

   &:focus {
      border-bottom: 1px solid #1a1a1a;
   }
`;

