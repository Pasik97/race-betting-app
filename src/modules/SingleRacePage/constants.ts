import { RouteComponentProps } from "react-router-dom";

interface SingleRacePageParams {
   id: string;
}

export type SingleRacePageProps = RouteComponentProps<SingleRacePageParams>;
