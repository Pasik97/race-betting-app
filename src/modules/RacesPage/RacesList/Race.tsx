import React from 'react';
import * as P from './parts';
import * as TP from 'components/BasicTable/parts';
import { RaceWithBet } from 'store/races/constants';

interface RaceProps {
   race: RaceWithBet;
}

const Race: React.FC<RaceProps> = ({ race }) => (
   <TP.Row key={race.id}>
      <TP.Cell><P.StyledLink to={'/races/' + race.id}>{race.id}</P.StyledLink></TP.Cell>
      <TP.Cell><P.StyledLink to={'/races/' + race.id}>{race.name}</P.StyledLink></TP.Cell>
      <TP.Cell><TP.Label>{race.active ? 'Active' : 'Inactive'}</TP.Label></TP.Cell>
   </TP.Row>
);

export default Race;
