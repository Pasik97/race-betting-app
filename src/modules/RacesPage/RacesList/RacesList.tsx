import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import * as P from './parts';
import Race from './Race';
import * as TP from 'components/BasicTable/parts';
import { getRaces, getRacesOrder } from 'store/races/selectors';

const RacesList: React.FC = () => {
   const races = useSelector(getRaces);
   const order = useSelector(getRacesOrder);

   const [shouldDisplayActive, setShouldDisplayActive] = useState<boolean>(false);

   const onCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
      setShouldDisplayActive(event.target.checked);
   }

   const filteredRacesOrder = !shouldDisplayActive ? order : order?.filter((id) => races[id].active);

   return !order.length ? null : (
      <P.RacesListWrapper>
         <P.CheckboxWrapper>
            <TP.Label>Show only active races:</TP.Label>
            <P.Checkbox
               type="checkbox"
               name="activeRaces"
               checked={shouldDisplayActive}
               onChange={onCheckboxChange}
            />
         </P.CheckboxWrapper>
         <TP.Row>
            <TP.HeaderCell>Id</TP.HeaderCell>
            <TP.HeaderCell>Race Name</TP.HeaderCell>
            <TP.HeaderCell>Status</TP.HeaderCell>
         </TP.Row>
         {filteredRacesOrder?.map((id) => <Race key={id} race={races[id]} />)}
      </P.RacesListWrapper>
   );
};

export default RacesList;
