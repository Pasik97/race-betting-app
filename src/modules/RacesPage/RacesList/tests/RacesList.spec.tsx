import { act, create } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import * as M from './mockData';
import RacesList from "../RacesList";
import { Checkbox } from "../parts";
import { mockRacesState } from "store/races/tests/mockData";
import { initilRacesState } from "store/races/constants";

jest.mock('../Race');

describe('<RacesList />', () => {
   it('should render checkbox and races list', () => {
      const component = create(
         <BrowserRouter>
            <Provider store={M.createMockRaceStore(mockRacesState)}>
               <RacesList />
            </Provider>
         </BrowserRouter>,
      );

      expect(component.toJSON()).toMatchSnapshot();
   });

   it('should render only active races if shouldDisplayActive is set to true', () => {
      const component = create(
         <BrowserRouter>
            <Provider store={M.createMockRaceStore(mockRacesState)}>
               <RacesList />
            </Provider>
         </BrowserRouter>,
      );

      const checkbox = component.root.findByType(Checkbox);

      act(() => {
         checkbox.props.onChange({ target: { checked: true } });
      });

      expect(component.toJSON()).toMatchSnapshot();
   });

   it('should render null if races order is an empty array', () => {
      const component = create(
         <BrowserRouter>
            <Provider store={M.createMockRaceStore(initilRacesState)}>
               <RacesList />
            </Provider>
         </BrowserRouter>,
      );

      expect(component.toJSON()).toBeNull();
   });

   describe('onCheckboxChange', () => {
      it('should set checkbox value', () => {
         const component = create(
            <BrowserRouter>
               <Provider store={M.createMockRaceStore(mockRacesState)}>
                  <RacesList />
               </Provider>
            </BrowserRouter>,
         );
         let checkbox = component.root.findByType(Checkbox);

         act(() => {
            checkbox.props.onChange({ target: { checked: true } });
         });

         checkbox = component.root.findByType(Checkbox);

         expect(checkbox.props.checked).toEqual(true);

         act(() => {
            checkbox.props.onChange({ target: { checked: false } });
         });

         checkbox = component.root.findByType(Checkbox);

         expect(checkbox.props.checked).toEqual(false);
      });
   });
});
