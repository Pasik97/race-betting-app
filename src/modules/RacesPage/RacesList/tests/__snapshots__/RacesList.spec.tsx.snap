// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RacesList /> should render checkbox and races list 1`] = `
<div
  className="sc-bdvvtL"
>
  <div
    className="sc-dkPtRN eAHUEW"
  >
    <p
      className="sc-iCfMLu iDUcVZ"
    >
      Show only active races:
    </p>
    <input
      checked={false}
      className="sc-hKwDye deOSNL"
      name="activeRaces"
      onChange={[Function]}
      type="checkbox"
    />
  </div>
  <div
    className="sc-gKclnd huABHg"
  >
    <div
      className="sc-eCImPb sc-jRQBWg jHOpYB enxhMK"
    >
      Id
    </div>
    <div
      className="sc-eCImPb sc-jRQBWg jHOpYB enxhMK"
    >
      Race Name
    </div>
    <div
      className="sc-eCImPb sc-jRQBWg jHOpYB enxhMK"
    >
      Status
    </div>
  </div>
  (Race /)
  (Race /)
</div>
`;

exports[`<RacesList /> should render only active races if shouldDisplayActive is set to true 1`] = `
<div
  className="sc-bdvvtL"
>
  <div
    className="sc-dkPtRN eAHUEW"
  >
    <p
      className="sc-iCfMLu iDUcVZ"
    >
      Show only active races:
    </p>
    <input
      checked={true}
      className="sc-hKwDye deOSNL"
      name="activeRaces"
      onChange={[Function]}
      type="checkbox"
    />
  </div>
  <div
    className="sc-gKclnd huABHg"
  >
    <div
      className="sc-eCImPb sc-jRQBWg jHOpYB enxhMK"
    >
      Id
    </div>
    <div
      className="sc-eCImPb sc-jRQBWg jHOpYB enxhMK"
    >
      Race Name
    </div>
    <div
      className="sc-eCImPb sc-jRQBWg jHOpYB enxhMK"
    >
      Status
    </div>
  </div>
  (Race /)
</div>
`;
